synedrion/src/tools/sss.rs

```diff
@@ -33,23 +34,25 @@ fn evaluate_polynomial<T>(coeffs: &[T], x: &Scalar) -> T
 where
     T: Copy + Add<T, Output = T> + for<'a> Mul<&'a Scalar, Output = T>,
 {
+    assert!(coeffs.len() > 1, "Expected coefficients to be non-empty");
     // Evaluate in reverse to save on multiplications.
     // Basically: a0 + a1 x + a2 x^2 + a3 x^3 == (((a3 x) + a2) x + a1) x + a0
-    let mut res = coeffs[coeffs.len() - 1];
-    for i in (0..(coeffs.len() - 1)).rev() {
-        res = res * x + coeffs[i];
-    }
-    res
+
+    let (acc, coeffs) = coeffs.split_last().expect("Coefficients is not empty");
```

Comment:
I probably wouldn't bother and just added an exception for clippy here :) But since you did, I guess we can keep it.